readme.md

```diff
@@ -15,834 +14,1113 @@
 
+```javascript
+CKEDITOR.config.myPlugin_myProperty = true;
+```
 
-    A. You are not a human code compiler/compressor, so don't try to be one.
+## 2.7. Faces of `this`
 
-    The following code is an example of egregious naming:
+### 2.7.1. Event listeners
 
-    ```javascript
+Keep `this` "local" to the class. This means &ndash; do not bind e.g. a function in `selection.js` file to editor instance and do not create a function which should be ran in editor context. This way you'll avoid confusion what `this` refers to.
 
-    // 6.A.1.1
-    // Example of code with poor names
+```javascript
+function Selection() {
+	var editor = this.editor;
 
-    function q(s) {
-        return document.querySelectorAll(s);
-    }
-    var i,a=[],els=q("#foo");
-    for(i=0;i<els.length;i++){a.push(els[i]);}
-    ```
+	// Bad:
+	editor.on( 'setData', someCallback, editor );
 
-    Here's the same piece of logic, but with kinder, more thoughtful naming (and a readable structure):
+	// Still bad:
+	editor.on( 'setData', CKEDITOR.tools.bind( someCallback, editor ) );
 
-    ```javascript
+	// Good:
+	editor.on( 'setData', someCallback, this );
 
-    // 6.A.2.1
-    // Example of code with improved names
+	// Good:
+	this.getEditor().on( 'setData', function( evt ) {
+		evt.editor.doSomething();
+	} );
 
-    function query( selector ) {
-      return document.querySelectorAll( selector );
-    }
+	// Good too:
+	var that = this;
+	editor.on( 'setData', function( evt ) {
+		editor.setCurrentSelection( that );
+	} );
+}
 
-    var id = 0,
-        elements = [],
-        matches = query( '#foo' ),
-        length = matches.length;
+function someCallback() {
+	// Should refers to Selection, not Editor.
+	this;
+}
+```
 
-    for ( ; id < length; id++ )
-        elements.push( matches[ id ] );
+### 2.7.2. Use `thisArg` argument
 
-    ```
+Several prototype methods of ES 5.1 built-ins come with a special `thisArg` signature, which should be used whenever possible. CKEditor follows this practice too, so for example [`eventTarget.on`](http://docs.ckeditor.com/#!/api/CKEDITOR.event-method-on) also accepts context as an argument.
 
-    A few additional naming pointers:
+```javascript
+var obj = { f: 'foo', b: 'bar', q: 'qux' };
 
-    ```javascript
+Object.keys( obj ).forEach( function( key ) {
 
-    // 6.A.3.1
-    // Naming strings
+  // |this| now refers to `obj`
 
-    `dog` is a string
+  console.log( this[ key ] );
 
+}, obj ); // <-- the last arg is `thisArg`
 
-    // 6.A.3.2
-    // Naming arrays
+// Prints...
 
-    `dogs` is an array of `dog` strings
+// 'foo'
+// 'bar'
+// 'qux'
+```
 
+`thisArg` can be used with `Array.prototype.every`, `Array.prototype.forEach`, `Array.prototype.some`, `Array.prototype.map`, `Array.prototype.filter`.
```

Comment:
Keep in mind that all of those are not working on IE8 and we have polyfills for them, so maybe refer to them instead.